control-plane-operator/controllers/hostedcontrolplane/storage/envreplace.go

```diff
@@ -62,10 +66,11 @@ func (er *environmentReplacer) setOperatorImageReferences(images map[string]stri
 	// `images` is map from payload image name -> image URL
 	// Create map from env. var name -> image URL
 	for envVar, payloadName := range operatorImageRefs {
-		// TODO: *_DRIVER_IMAGE and LIVENESS_PROBE_IMAGE for csi-driver-node should be separated
-		// from csi-driver-controller and looked up from the userImages
-		// strings.HasSuffix(envVar, "_DRIVER_IMAGE") || envVar == "LIVENESS_PROBE_IMAGE"
-		if envVar == "NODE_DRIVER_REGISTRAR_IMAGE" {
+		if strings.HasSuffix(envVar, "_CONTROL_PLANE_IMAGE") {
```

Comment:
I don't like some control plane env. vars having _CONTROL_PLANE_IMAGE suffix and others not. It should be consistent.

And if you want to continue with inconsistency, the conditions in the whole `if then else` look too complicated.
Would it be better to have a set of env. vars that are taken from `userImages` and the rest from `images`?

```
userImageEnvNames := sets.NewString("NODE_DRIVER_REGISTRAR_IMAGE", "LIVENESS_PROBE_IMAGE", "AWS_EBS_DRIVER"
if userImageEnvNames.Has(envVar) { ... }
```

Reply:
@jsafrane So the issue is that we are creating a net new envvar and we have to choose a name for it.

It isn't a matter of using the control plane or user release image for AWS_EBS_DRIVER, for example.  We need a new envvar so we can pass both the control plane and user release images for AWS_EBS_DRIVER.

Since the control plane specific envvar is the one specific to hypershift, I figured that would be the new one as to disturb the existing paths as little as possible.

I thought that using the _CONTROL_PLANE_IMAGE suffix both 1) created a convention for the feature naming of the variables if we decide to enable this for more platforms and 2) we don't have to change the code in this `if` block in the future; just add the envvar to `operatorImageRefs`.

Reply:
However, if you want to break `operatorImageRefs` into `controlPlaneImageRefs` and `userImageRefs`, and not rely on bespoke knowledge embedded in the `if` statement, I can do that